lib/common/grpc-service.js

```diff
@@ -446,26 +447,43 @@ GrpcService.createDeadline_ = function(timeout) {
 };
 
 /**
- * Checks for a grpc error code and extends the Error object with additional
+ * Checks for a grpc status code and extends the supplied object with additional
  * information.
  *
  * @private
  *
- * @param {error} err - The original request error.
- * @return {error|null}
+ * @param {object} obj - The object to be extended.
+ * @param {object} response - The grpc response.
+ * @return {object|null}
  */
-GrpcService.getError_ = function(err) {
-  if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
-    var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
+GrpcService.extendGrpcResponse_ = function(obj, response) {
```

Comment:
Our confusion speaks to the confusing nature of what the code is doing, so I just want to make things as clear as possible.

If all we are really doing is decorating the response from gRPC, we should just *always* do it by calling `decorateGrpcResponse_(grpcResponse)`. If we want to know if it's an error, so we can treat it as an error, we should check if it's an error:

```js
var decoratedGrpcResponse = GrpcService.decorateGrpcResponse_(resp);

if (is.error(decoratedGrpcResponse)) {
  callback(decoratedGrpcResponse);
}
```

When we want to emit the response:

```js
.on('status', function(grpcResponse) {
  var decoratedGrpcResponse = GrpcService.decorateGrpcResponse_(resp);
  this.emit('response', decoratedGrpcResponse);
});
```